pkg/cloudprovider/providers/openstack/openstack_loadbalancer.go

```diff
@@ -560,6 +562,46 @@ func getNodeSecurityGroupIDForLB(compute *gophercloud.ServiceClient, nodes []*v1
 	return nodeSecurityGroupIDs.List(), nil
 }
 
+// getFloatingNetworkIdForLB returns a floating-network-id for cluster.
+func getFloatingNetworkIdForLB(client *gophercloud.ServiceClient) (string, error) {
+	var FloatingNetworkIds []string
+
+	type NetworkWithExternalExt struct {
+		networks.Network
+		external.NetworkExternalExt
+	}
+	var externalNetwork []NetworkWithExternalExt
+	err := networks.List(client, networks.ListOpts{}).EachPage(func(page pagination.Page) (bool, error) {
+		err := networks.ExtractNetworksInto(page, &externalNetwork)
```

Comment:
Move the `externalNetwork` declaration 2 lines above to immediately above this line (_inside_ this pagination callback function).  That will help make sure nobody tries to use the value incorrectly outside/after this function.

Reply:
Done.